core/src/main/java/io/neow3j/protocol/rx/JsonRpc2_0Rx.java

```diff
@@ -31,32 +28,50 @@ public JsonRpc2_0Rx(Neow3j neow3j, ScheduledExecutorService scheduledExecutorSer
         this.scheduler = Schedulers.from(scheduledExecutorService);
     }
 
-    public Observable<BigInteger> neoBlockObservable(long pollingInterval) {
-        return Observable.create(subscriber -> {
-            BlockPolling blockPolling = new BlockPolling(neow3j, subscriber::onNext);
-            blockPolling.run(scheduledExecutorService, pollingInterval);
-            subscriber.setDisposable(Disposables.fromAction(blockPolling::cancel));
-        });
+    /**
+     * Creates an observable that emits new block index as they are produced by the Neo blockchain. The observable
+     * polls the Neo node in the given {@code pollingInterval} to check for the latest block index and emits all
+     * indexes since the last time it polled.
+     *
+     * @param pollingInterval The polling interval in milliseconds.
+     * @return the block index observable.
+     */
+    public Observable<BigInteger> blockIndexObservable(long pollingInterval) {
+        return Observable.create(subscriber ->
+                new BlockIndexPolling().run(neow3j, subscriber, scheduledExecutorService, pollingInterval)
+        );
```

Comment:
This is one of the most severe changes. Basically, the only reason why we need an `BlockIndexPolling` object and cannot do this without any object state is the `currentBlockIdx` in `BlockIndexPolling`. It is required to know the block index up to which the observable already emitted. All other state/variables of `BlockIndexPolling` are not necessary as variables so I moved them into the `run` method directlly.

Reply:
Can you explain to me, in a pair programming session tomorrow? Because I'm sure that the old implementation didn't need to keep track of state since the reactive way could already implicitly hold "where it stopped". As far as I remember, that was the magic of why it worked so far.
Am I wrong?

Reply:
yes, let's do a pair programming session tomorrow. It's best if you stay very sceptic about my changes. In the end even if my changes are erroneous I will at least have learned something.